File: Sesion 6/Marco teorico/9QUÉ ES UN MIEMBRO STATIC/9.py
class MiClase:
    contador = 0

    def __init__(self):
        MiClase.contador += 1

    @staticmethod
    def mostrar_contador():
        print("El contador es:", MiClase.contador)


obj1 = MiClase()
obj2 = MiClase()
obj3 = MiClase()

MiClase.mostrar_contador()

File: Sesion 6/Marco teorico/9QUÉ ES UN MIEMBRO STATIC/9.java
class MiClase {
    static int contador = 0;

    MiClase() {
        contador++;
    }

    static void mostrarContador() {
        System.out.println("El contador es: " + contador);
    }
}

public class Main {
    public static void main(String[] args) {
        MiClase obj1 = new MiClase();
        MiClase obj2 = new MiClase();
        MiClase obj3 = new MiClase();

        // Llamada al método estático desde la clase (no se necesita una instancia)
        MiClase.mostrarContador();
    }
}
